src/langchain_google_cloud_sql_pg/pgsql_loader.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import List, Optional, Iterator
+
+from langchain_core.documents import Document
+from langchain_community.document_loaders.base import BaseLoader
+from langchain.text_splitter import TextSplitter, RecursiveCharacterTextSplitter
+
+# TODO Move to PgSQLEngine Implementation from Vectorstore PR once it is merged
+from langchain_google_cloud_sql_pg.pgsql_engine import PgSQLEngine
+
+DEFAULT_METADATA_COL = "langchain_metadata"
+
+
+class PgSQLLoader(BaseLoader):
+    """Load documents from `CloudSQL Postgres`.
+
+    Each document represents one row of the result. The `content_columns` are
+    written into the `content_columns`of the document. The `metadata_columns` are written
+    into the `metadata_columns` of the document. By default, first columns is written into
+    the `page_content` and everything else into the `metadata`.
+    """
+
+    def __init__(
+        self,
+        engine: PgSQLEngine,
+        query: str,
+        table_name: str,
+        *,
+        content_columns: Optional[List[str]] = None,
+        metadata_columns: Optional[List[str]] = None,
+        format: Optional[str] = None,
+        read_only: Optional[bool] = None,
+        time_out: Optional[int]  = None,
+
+    ) -> None:
+        """Initialize CloudSQL Postgres document loader."""
+
+        self.engine = engine
+        self.table_name = table_name
+        self.query = query
+        self.content_columns = content_columns
+        self.metadata_columns = metadata_columns
+        self.format = format
+        self.read_only = read_only
+        self.time_out = time_out
+
+    # Partially Implemented
+    def load(self) -> List[Document]:
+        """Load CloudSQL Postgres data into Document objects."""
+        return self.alazy_load()
```

Comment:
```suggestion
        return list(self.alazy_load())
```
list() will be needed to convert the iterator to a list